CMSgov/managed-care-review
# fix: Complete and error summary in FileUpload list view

## Summary
Two bugs 
- error count is wrong
- complete count is also wrong

**6f2e54adfffde8c4713b46fa6405b28259278b52 is the actual bugfix**

Also made a change to rely just on one prop (`shouldDisplayMissingCategoriesError`) when FileUpload determines if/when categories are relevant.  The prop is just set to false in list view or when contract only submission.

#### Related issues
https://qmacbis.atlassian.net/browse/OY2-16435

#### Test cases covered

- Add tests to the "list summary" describe block in `FileUpload.test.tsx` for error count, complete count, and pending count.
- Add tests for error state and checkboxes still appearing when expected on Documents page after I broke that (accidentally removed the checkboxes when they were supposed to be there in error state) but tests still passed.

## QA guidance

Place both file upload list and file upload table in the different error states and check that count is correct. Can use storybook for quickly viewing error states for failed scan and failed upload.
